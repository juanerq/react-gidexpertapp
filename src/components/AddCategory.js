import { useState } from "react"
import PropTypes from 'prop-types'

const AddCategory = ({setCategory}) => {

  const [inputValue, setInputValue] = useState('')

  const handleInputValue = (e) => {
    setInputValue( e.target.value )
  }

  const handleSubmit = (e) => {
    e.preventDefault()

    if (inputValue.trim().length > 2) {
      setCategory( cats => [inputValue, ...cats])
      setInputValue('')
    }
  }

  return (
    <form onSubmit={ handleSubmit }>
      <p>{ inputValue }</p>
      <input 
        type='text'
        value={ inputValue }
        onChange={ handleInputValue }
      />
    </form>
  )
}

AddCategory.propTypes = {
  setCategory: PropTypes.func.isRequired
}

export default AddCategory