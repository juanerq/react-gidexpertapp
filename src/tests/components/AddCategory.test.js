import { shallow } from 'enzyme'
import AddCategory from '../../components/AddCategory'

describe('Component <AddCategory/>', () => {
  const setCategories = jest.fn()
  let wrapper = shallow( <AddCategory setCategory={ setCategories }/> )

  beforeEach(() => {
    jest.clearAllMocks()
    wrapper = shallow( <AddCategory setCategory={ setCategories }/> )
  })

  test('It should display AddCategory Item correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('Should change the text box', () => {

    const value = 'Todo bien'

    const input = wrapper.find('input')
    input.simulate('change', { target: { value } })

    expect( wrapper.find('p').text() ).toBe( value )

  })

  test('Should not post the information with submit', () => {

    wrapper.find('form').simulate('submit', { preventDefault(){} } )
    expect( setCategories ).not.toHaveBeenCalled()

  })

  test('Should call setcategories and clear the input', () => {

    const value = 'Todo bien' 

    const input = wrapper.find('input')
    input.simulate('change', { target: { value } })

    expect( wrapper.find('p').text() ).toBe( value )

    wrapper.find('form').simulate('submit', { preventDefault(){} })

    expect( setCategories ).toHaveBeenCalledTimes(1)
    expect( setCategories ).toHaveBeenCalledWith( expect.any(Function) )

    expect( input.prop('value').length ).toBe( 0 )

  })

})