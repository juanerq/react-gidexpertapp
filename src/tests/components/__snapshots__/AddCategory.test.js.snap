// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <AddCategory/> It should display AddCategory Item correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
